src/main/java/org/apache/sling/jcr/base/internal/LoginAdminAllowList.java

```diff
@@ -83,23 +75,22 @@ public class LoginAdminAllowList {
             cardinality = ReferenceCardinality.MULTIPLE,
             policy = ReferencePolicy.DYNAMIC,
             policyOption = ReferencePolicyOption.GREEDY
-    ) @SuppressWarnings("unused")
+    )
     void bindAllowListFragment(AllowListFragment fragment) {
         allowListFragments.add(fragment);
         LOG.info("AllowListFragment added '{}'", fragment);
     }
 
-    @SuppressWarnings("unused")
     void unbindAllowListFragment(AllowListFragment fragment) {
         allowListFragments.remove(fragment);
         LOG.info("AllowListFragment removed '{}'", fragment);
     }
 
-    @Activate @Modified @SuppressWarnings("unused")
-    void configure(LoginAdminAllowListConfiguration configuration, Map<String, Object> properties) {
-        this.config = new ConfigurationState(configuration);
-        ensureBackwardsCompatibility(properties, PROP_WHITELIST_BUNDLES_DEFAULT);
-        ensureBackwardsCompatibility(properties, PROP_WHITELIST_BUNDLES_ADDITIONAL);
+    @Activate @Modified
+    void configure(final LoginAdminAllowListConfiguration configuration, final Map<String, Object> properties) {
```

Comment:
What about using a second annotation, without metatype, like you did with the [`VanityPathConfigurer.DeprecatedVanityConfig`](https://github.com/apache/sling-org-apache-sling-resourceresolver/blob/a24a17467b334f9076f91c2dccd2491dfc9ae96b/src/main/java/org/apache/sling/resourceresolver/impl/ResourceResolverFactoryActivator.java#L221)?

Reply:
We could but we need the properties for all the methods below, so it is a little bit easier this way